docs/operations/repository_structure.md

```diff
@@ -0,0 +1,68 @@
+versioning, and automation. This applies to Plugin, Skill, Service, and Library
+repositories as [defined here](https://neongeckocom.github.io/neon-docs/overview/definitions/).
+
+## Default Branches
+Repositories always contain `dev` and `master` branches.
+
+| Branch | Version | GitHub Release | Docker Image |
+|--------|---------|----------------|--------------|
+| dev    | alpha   | Pre-Release    | dev          |
+| master | stable  | Release        | latest       |
+
+New code is always PR'd into the `dev` branch as discrete changes. This 
+[Alpha Release Workflow](https://neongeckocom.github.io/neon-docs/operations/libraries/workflow_alpha_release/)
+describes the process of making changes to a repository. This allows the commit
+history of the `dev` branch to be used as a changelog for the project, with each
+change corresponding to a specific (alpha) version and a descriptive commit/PR.
+
+When an alpha release is ready to be promoted to a satble release, a PR is created
+to merge changes from `dev` into `master` with a version number that generally 
+follows [semver](https://semver.org/) or [calver](https://calver.org/). This
+[Stable Release Workflow](https://neongeckocom.github.io/neon-docs/operations/libraries/workflow_stable_release/)
+describes the process of proposing and approving a stable release. A commit with
+the specific version number is created on the `master` branch, along with a GitHub
+Release so `master` always reflects the code from the latest release.
+
+## Versioning
+Repositories always publish alpha/beta/pre-release versions and latest/stable/normal 
+versions. The terms are often used interchangeably, but this document will
+follow SemVer's terminology of "pre-release" and "normal release".
+
+Pull requests to `dev` and pre-releases map 1:1, so a merged PR to `dev` 
+corresponds to a specific pre-release and vice versa. Pull requests to `master`
+map to normal releases in the same way.
+
+A pre-release should always have a corresponding GitHub Pre-release and a normal
+release should have a GitHub Release.
+
+For packages published to PyPI, the PyPI version will always match the repository
+version. Docker images that correspond to a specific version will always have a
+tag matching the repository version.
+
+## GitHub Releases
+GitHub pre-release tags will always correspond to a specific "pre-release" and 
+releases tags will always correspond to a specific "normal release".
+> An exception to this is the Neon OS repository which will list a normal release
+  as a pre-release temporarily so the release may be manually validated before
+  being made available to users tracking the stable update track.
+
+## Docker Containers
+For repositories that publish Docker images, the `dev` image tag will always
+match the `dev` branch and the `latest` image will match the `master` branch.
+Images that correspond to any release (pre-release or normal release) will have
+an additional tag indicating the version.
```

Comment:
Consider setting `stable` to master as well. It's a practice I've seen in a few Docker repositories that should really be more widespread. Even if it tracks directly with `latest`, it's much more descriptive, and makes it clear to users who haven't read this document that this image is not tracking anything alpha.

Reply:
Is `stable` a special tag for Docker like `latest`, or just a commonly used one? I have no objection to adding it; as you noted, it is much more descriptive

Reply:
It's neither special to Docker nor commonly used, but every time I see it I'm happy it's there 🙂 It's a good habit to get into, I think, and borrow from OS packages